pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -0,0 +1,144 @@
+                                  recurse=True):
+        """Internal API for setting up bind mounts between the physical root and
+           sysroot, also ensures we track them in self._internal_mounts so we can
+           cleanly unmount them.
+
+           :param src: Source path, will be prefixed with physical or sysroot
+           :param dest: Destination, will be prefixed with sysroot (defaults to same as src)
+           :param src_physical: Prefix src with physical root
+           :param bind_ro: Make mount read-only
+           :param recurse: Use --rbind to recurse, otherwise plain --bind
+        """
+        # Default to the same basename
+        if dest is None:
+            dest = src
+        # Almost all of our mounts go from physical to sysroot
+        if src_physical:
+            src = conf.target.physical_root + src
+        else:
+            src = conf.target.system_root + src
+        # Canonicalize dest to the full path
+        dest = conf.target.system_root + dest
+        if bind_ro:
+            safe_exec_with_redirect("mount", ["--bind", src, src])
+            safe_exec_with_redirect("mount", ["--bind", "-o", "remount,ro", src, src])
+        else:
+            # Recurse for non-ro binds so we pick up sub-mounts
+            # like /sys/firmware/efi/efivars.
+            if recurse:
+                bindopt = '--rbind'
+            else:
+                bindopt = '--bind'
+            safe_exec_with_redirect("mount", [bindopt, src, dest])
+        self._internal_mounts.append(src if bind_ro else dest)
+
+    def run(self):
+        """Run the task.
+
+        :return: list of bindmounts created for internal use
+        :rtype: list of str
+        """
+        device_tree = STORAGE.get_proxy(DEVICE_TREE)
+        mount_points = device_tree.GetMountPoints()
+
+        # Currently, blivet sets up mounts in the physical root.
+        # We used to unmount them and remount them in the sysroot, but
+        # since 664ef7b43f9102aa9332d0db5b7d13f8ece436f0 we now just set up
+        # bind mounts.
+
+        # Make /usr readonly like ostree does at runtime normally
+        self._setup_internal_bindmount('/usr', bind_ro=True, src_physical=False)
+
+        # Explicitly do API mounts; some of these may be tracked by blivet, but
+        # we'll skip them below.
+        api_mounts = ["/dev", "/proc", "/run", "/sys"]
+        for path in api_mounts:
+            self._setup_internal_bindmount(path)
+
+        # Handle /var; if the admin didn't specify a mount for /var, we need
+        # to do the default ostree one.
+        # https://github.com/ostreedev/ostree/issues/855
+        var_root = '/ostree/deploy/' + self._source_config.osname + '/var'
+        if mount_points.get("/var") is None:
+            self._setup_internal_bindmount(var_root, dest='/var', recurse=False)
+        else:
+            # Otherwise, bind it
+            self._setup_internal_bindmount('/var', recurse=False)
```

Comment:
I would like to break these into methods a bit more. Something like
``handle_mount_points``
and inside of that calls of
``handle_var_mount_points``
``handle_other_mount_points``

Or something similar. Plenty of these comments could then be as doc string.